### PR TITLE
Fix relationship metrics

### PR DESCRIPTION
**Pull request by xithan**
#152277747

### CODE REVIEW

File: mod/caching/set/type_plus_right/wikirate_company/wikirate_topic.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@ def company_name
   Answer.select(:company_id).where(metric_id: metric_id).uniq
         .pluck(:company_id).map do |company_id|
     # faster way to get this from company+topic?
-    Card.fetch(company_id.cardname.trait :wikirate_topic)
+    Card.fetch(company_id.cardname.trait(:wikirate_topic))
```
Comment (ethn) [inline mod/caching/set/type_plus_right/wikirate_company/wikirate_topic.rb:20]: do we have a style preference for these situations?

File: mod/csv_import/set/abstract/import/01_table_row.rb [hunk 2]
```diff
@@ -104,7 +103,7 @@ def respond_to_missing? method, _include_private=false
   end
 
   def field_from_method method
-    method =~ /^(\w+)_field$/ ? $1 : nil
+    method =~ /^(\w+)_field$/ ? Regexp.last_match(1) : nil
```
Comment (ethn) [inline mod/csv_import/set/abstract/import/01_table_row.rb:106]: I'm not in love with last_match, but I can't figure out how to do this common pattern much more cleanly.

```
m = method.match /^(\w+)_field$/
m && m[1]
```
I wish this worked: `method.match(/^(\w+)_field$/)&[1]

**Review by ethn — APPROVED** [review]
Whenever there are this many changes, any help getting to the meat of the main logic changes would help.  I skimmed the whole thing, and the relationship changes look good from what I can see.

File: mod/caching/set/type_plus_right/wikirate_company/wikirate_topic.rb (hunk 1 above)
Comment (xithan) [reply mod/caching/set/type_plus_right/wikirate_company/wikirate_topic.rb:20]: inner parenthesis are mandatory, outer ones are optional

File: mod/csv_import/set/abstract/import/01_table_row.rb (hunk 2 above)
Comment (xithan) [reply mod/csv_import/set/abstract/import/01_table_row.rb:106]: how about `method.match(/^(\w+)_field$/)&.captures&.at(0)`?

**Comment by xithan** [comment]
You can use the commit links to get rid of all the clean up changes.

File: mod/csv_import/set/abstract/import/01_table_row.rb (hunk 2 above)
Comment (ethn) [reply mod/csv_import/set/abstract/import/01_table_row.rb:106]: it's actually more readable than my "wish" but probably less so than either the current code or my attempt. 

We could also add our own method to String:
```
def nth_match n, regexp
...

method.nth_match 1, /^(\w+)_field$/
```

Comment (xithan) [reply mod/csv_import/set/abstract/import/01_table_row.rb:106]: I don't like that you first say in what part of the match you are interested in and then you define and do the match. 
We could add `def capture n` to MatchData instead:
`method.match(/^(\w+)_field$/)&.capture(0)`

Comment (xithan) [reply mod/csv_import/set/abstract/import/01_table_row.rb:106]: https://github.com/decko-commons/decko/pull/129/files

**Pull request merged**